### PR TITLE
always pass EPEK to enclave wether it remembers auth method or not

### PR DESCRIPTION
**Pull request by Mohammed-Mamoun98**
EPEK 👉 Expected public Encryption Key

### CODE REVIEW

File: apps/idos-enclave/src/pages/App.tsx [hunk 1]
```diff
@@ -102,11 +102,11 @@ export function App({ store, enclave }: AppProps) {
       throw new Error(`Unexpected request from parent: ${requestData.intent}`);
 
     responsePort.current = ports[0];
+    setEncryptionUserPublicKey(event.data.message.expectedUserEncryptionPublicKey);
```
Comment (juliosantos) [inline apps/idos-enclave/src/pages/App.tsx:105]:
```suggestion
    setEncryptionUserPublicKey(requestData.message?.expectedUserEncryptionPublicKey);
```
This seems unsafe just looking at the rest of the file, wdyt of this alt?

Comment (juliosantos) [reply apps/idos-enclave/src/pages/App.tsx:105]: Also, how does `setEncryptionUserPublicKey` handle empty/null/undefined values?

Comment (Mohammed-Mamoun98) [reply apps/idos-enclave/src/pages/App.tsx]: we can either throw an error "No Public encryption key found" OR just log and error with same message. what do u think?

Comment (pkoch) [reply apps/idos-enclave/src/pages/App.tsx]: Throw. Let's make it loudly blow up.

File: apps/idos-enclave/src/pages/App.tsx [hunk 2]
```diff
@@ -102,11 +102,11 @@ export function App({ store, enclave }: AppProps) {
       throw new Error(`Unexpected request from parent: ${requestData.intent}`);
 
     responsePort.current = ports[0];
+    setEncryptionUserPublicKey(requestData.message.expectedUserEncryptionPublicKey);
```
Comment (juliosantos) [inline apps/idos-enclave/src/pages/App.tsx:105]:
```suggestion
    setEncryptionUserPublicKey(requestData.message?.expectedUserEncryptionPublicKey);
```

We're using safe navigation just a few lines below, should't we do the same here?

**Review by juliosantos — APPROVED** [review]

**Pull request merged**